### ISSUE TITLE
items that my update more then 1 times(metrics in the past) handles inproperly

### ISSUE THREAD

**Issue by tantra35** [moira-alert/cache#1]
When we use amazon cloud watch, it doesn't every time give metrics correctly(less that it must be), so we correct them time to time, and this is mean that the the same metrics may come 2 and more times, with the same timestamp. But moira cache do follow:

```
metricValue := fmt.Sprintf("%v %v", m.value, m.timestamp)
c.Send("ZADD", metricKey, m.timestamp, metricValue)
```

So for the same value(with the same timestamp) we have 2 and more different values, like this;

```
81) "1123 1452634560"
82) "679 1452634560"
```

In this situation moira doesn't behave like graphite, it doens't allow set point in the past, and doesn't allow update metric

The simeplest workaround will be, remove alredy existens elements by score:

```
c.Send("ZREMRANGEBYSCORE", metricKey, m.timestamp, m.timestamp)
```

**Comment by gmlexx** [comment]
Can you try your workaround and tell that it really helps?
I'm concern about performance, because "zadd" is about 50% of all redis load.
If we just add "zremrangebyscore" every time, load may grow significantly.
Anyway, it can be optional.

**Comment by tantra35** [comment]
We alredy use it, aboyt 10 hours, now all ok. Use "zremrangebyscore" is simple work around(for faster fix for us), and needed more complex solution, that change worker-checker and cache data structures. load of redis isn't probleam we alltime cant scale it to more machines

**Comment by gmlexx** [comment]
Possible good solution is to replace ZADD to "HSET timestamp value",
where timestamp is HSET key and value is HSET value.

So, insert operation will be much faster and it fix this problem.

Metrics cleanup will be more complex, but still fast in case of few hours of TTL interval.

**Comment by gmlexx** [comment]
Experiment with HSET shows that worker performance will degradate 3x times, so we did other solution,
Metric value format changed from "value timestamp" to "timestamp value".
This will lead to fetching metrics from redis zrangebyscore in the timestamp order, so the last value will override previous.

**Comment by gmlexx** [comment]
Fixed in v1.1.1
